src/osx/cocoa/dataview.mm

```diff
@@ -1985,8 +1985,13 @@ -(BOOL) becomeFirstResponder
 {
     // initialize scrollview (the outline view is part of a scrollview):
     NSScrollView* scrollview = (NSScrollView*) GetWXWidget();
+    wxBorder border = (wxBorder)(style & wxBORDER_MASK);
+
+    if ( (border == wxBORDER_DEFAULT) || (border == wxBORDER_THEME) )
+        [scrollview setBorderType:NSBezelBorder];
+    else
+        [scrollview setBorderType:NSNoBorder];
```

Comment:
I still think that we should use `wxWindow::GetBorder()` here and that we should extract this into some reusable function (`OSXApplyBorder()`?).

Moreover, as Stefan explained, the rest of the code uses `setBordered:NO`, so I'd rather do this here too for consistency.

Combining both of these thoughts, maybe this could be replaced with just

```cpp
if ( peer->GetBorder() == wxBORDER_NONE )
    [scrollview setBordered:NO];
```

? Or do we need to call `setBezeled:YES` too?

Reply:
Bordered is different than bezel type in this case. Bordered is hiding the bezel, but we still need to set the type.

Reply:
`setBorderType` and `setBordered` do different things and not all controls respond to both calls, there isn't a way to make it 100% consistent.

Reply:
Oh. Thanks, I hadn't realized that `setBorderType:NSNoBorder` and `setBordered:NO` could be doing different things.

Then I'd just like to extract this into `wxWindow::OSXApplyBorder(NSView* v)` and call it from all the places where it's needed.

Reply:
So the helper function should probably be called `OSXApplyBorderType()`...

Reply:
What do you have in mind for that function? Currently the macOS controls translate the wxBORDER styles passed to what makes sense for the control type (or ignored if they don't make sense).  Trying to abstract that into a helper function doesn't seem particularly useful since each control is very specific.  However, perhaps, abstracting the above scrollview styling into a helper function might make more sense since the styling needs to be applied to several scrollviews in the same way.

Reply:
That makes sense, however I'm not sure I'm familiar enough with the codebase to make all those changes.

Reply:
The new function can be added to `include/wx/osx/window.h` (it would be called as `peer->OSXApplyBorderType(view)` from the places where you're doing it now), AFAICS this should be simple to do, but please let me know if I'm missing something?

I could do it myself, but I wouldn't be able to write a good comment for it, which is what I'm after!

Reply:
I've been poking around with this and adding a function to window.h and then passing in the NSView results in a lot of `[view isKindOfClass:]` calls to determine what type of control it is to then apply the correct border style to. I'm not sure it's worth doing this way when we know the type of the view already at creation time and can apply it there which it is doing already. Thoughts? Comments?

Reply:
Sorry, I didn't mean to ask you to do anything complicated, i.e. write a function that would work for all kinds of windows. I really just wanted to perform a trivial refactoring and put the code you added into a function that would be only used for the views that need `setBorderType:` to be called and call it for them only.

Reply:
AH! Well, that would certainly be easier. :) OK, will do that.